### DISCUSSION TITLE
Multiple mutations in one request

### DISCUSSION THREAD

**Discussion by cleanoxi** [graphql-java-kickstart/graphql-spring-boot#423] [Q&A] [answered]
I have multiple mutations of the same kind that I want to execute in a single transaction. Can I do this?

`type Mutation {
    editAuthor(author: String!, book: String!): Boolean
}

mutation updateDetails {
    editAuthor(author: "Stephen King", book: "Castle Rock")
    editAuthor(author: "Kurt Vonnegut", book: "Cat's Cradle")
}`

**Comment by BlasiusSecundus** [comment, marked as answer]
Yes, you can do that using aliases: https://graphql.org/learn/queries/#aliases

```graphql
mutation updateDetails {
    king: editAuthor(author: "Stephen King", book: "Castle Rock")
    vonnegut: editAuthor(author: "Kurt Vonnegut", book: "Cat's Cradle")
}
```

Note that, however, this will not be a single transaction (one may succeed or fail independently from the other).
